driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -125,9 +148,10 @@ public void remove() {
                 if (!rawDocument.containsKey("_id")) {
                     throw new MongoChangeStreamException("Cannot provide resume functionality when the resume token is missing.");
                 }
-                resumeToken = rawDocument.getDocument("_id");
                 results.add(rawDocument.decode(changeStreamOperation.getDecoder()));
             }
+            // Should there be no postBatchResumeToken, cache the _id of the last document.
+            resumeToken = rawDocuments.get(rawDocuments.size() - 1).getDocument("_id");
```

Comment:
This will throw if `rawDocuments` is empty, which is an expected outcome in some circumstances.  I'm surprised no tests catch that.

Reply:
In `QueryBatchCursor#initFromQueryResult`, if there are no results, `nextBatch` is set to null, which would be returned from `next()` and `tryNext()`. This statement will not be reached if this is the case.

Reply:
Got it.